integration-tests/deployment/devenv/don.go

```diff
@@ -0,0 +1,266 @@
+	}
+	return err
+}
+
+type DON struct {
+	Nodes []Node
+}
+
+func (don *DON) NodeIds() []string {
+	var nodeIds []string
+	for _, node := range don.Nodes {
+		nodeIds = append(nodeIds, node.NodeId)
+	}
+	return nodeIds
+}
+
+func (don *DON) CreateSupportedChains(ctx context.Context, chains []ChainConfig) error {
+	var err error
+	for _, node := range don.Nodes {
+		if err1 := node.CreateCCIPOCR2SupportedChains(ctx, chains); err1 != nil {
+			err = multierror.Append(err, err1)
+		}
+	}
+	return err
+}
+
+func NewRegisteredDON(ctx context.Context, nodeInfo []NodeInfo, jd JobDistributor) (*DON, error) {
+	don := &DON{
+		Nodes: make([]Node, 0),
+	}
+	for i, info := range nodeInfo {
+		if info.Name == "" {
+			info.Name = fmt.Sprintf("node-%d", i)
+		}
+		node, err := NewNode(info)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create node %d: %w", i, err)
+		}
+		// node Labels so that it's easier to query them
+		if info.IsBootstrap {
+			// create multi address for OCR2, applicable only for bootstrap nodes
+			node.multiAddr = info.CLConfig.URL
+			// no need to set admin address for bootstrap nodes, as there will be no payment
+			node.adminAddr = ""
+			node.labels = append(node.labels, &ptypes.Label{
+				Key:   NodeLabelType,
+				Value: pointer.ToString(NodeLabelBootstrap),
+			})
+		} else {
+			// multi address is not applicable for non-bootstrap nodes
+			// explicitly set it to empty string to denote that
+			node.multiAddr = ""
+			node.labels = append(node.labels, &ptypes.Label{
+				Key:   NodeLabelType,
+				Value: pointer.ToString(NodeLabelPlugin),
+			})
+		}
+		// Set up Job distributor in node and register node with the job distributor
+		err = node.SetUpAndLinkJobDistributor(ctx, jd)
+		if err != nil {
+			return nil, fmt.Errorf("failed to set up job distributor in node %s: %w", info.Name, err)
+		}
+
+		don.Nodes = append(don.Nodes, *node)
+	}
+	return don, nil
+}
+
+func NewNode(nodeInfo NodeInfo) (*Node, error) {
+	gqlClient, err := client.New(nodeInfo.CLConfig.URL, client.Credentials{
+		Email:    nodeInfo.CLConfig.Email,
+		Password: nodeInfo.CLConfig.Password,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create FMS client: %w", err)
+	}
+	return &Node{
+		gqlClient: gqlClient,
+		name:      nodeInfo.Name,
+		adminAddr: nodeInfo.AdminAddr,
+	}, nil
+}
+
+type Node struct {
+	NodeId    string
+	JDId      string
+	gqlClient client.Client
+	labels    []*ptypes.Label
+	name      string
+	adminAddr string
+	multiAddr string
+}
+
+// CreateCCIPOCR2SupportedChains creates JobDistributorChainConfig for the node
+// it works under assumption that the node is already registered with the job distributor
+// expects bootstrap nodes to have type label set as bootstrap
+// It fetches the account address, peer id, OCR2 key bundle id and creates the JobDistributorChainConfig
+func (n *Node) CreateCCIPOCR2SupportedChains(ctx context.Context, chains []ChainConfig) error {
```

Comment:
since this is CCIP specific, should it really live in this package?

Reply:
not ccip specific

Reply:
Sorry, this is CCIP specific I will take the plugin input in func parameter